hack/generator/pkg/codegen/storage/type_converter.go

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package storage
+
+import (
+	"fmt"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// TypeConverter is used to create a storage variant of an API type
+type TypeConverter struct {
+	visitor           astmodel.TypeVisitor // Visitor used to achieve the required modification
```

Comment:
minor: Same question about comments to the side rather than above (and prefixed with var name)

Reply:
Moved.